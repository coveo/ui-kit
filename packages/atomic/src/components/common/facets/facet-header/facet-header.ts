import {FunctionalComponent} from '@/src/utils/functional-component-utils';
import {i18n} from 'i18next';
import {html, nothing} from 'lit';
import {when} from 'lit/directives/when.js';
import ArrowBottomIcon from '../../../../images/arrow-bottom-rounded.svg';
import ArrowTopIcon from '../../../../images/arrow-top-rounded.svg';
import CloseIcon from '../../../../images/close.svg';
import '../../atomic-icon/atomic-icon';
import {renderButton} from '../../button';
import {renderHeading} from '../../heading';

export interface FacetHeaderProps {
  i18n: i18n;
  label: string;
  numberOfActiveValues: number;
  isCollapsed: boolean;
  headingLevel: number;
  onToggleCollapse(): void;
  onClearFilters?(): void;
  headerRef?: (element?: HTMLButtonElement) => void;
}

export const renderFacetHeader: FunctionalComponent<FacetHeaderProps> = ({
  props,
}) => {
  const label = props.i18n.t(props.label);
  const expandFacet = props.i18n.t('expand-facet', {label});
  const collapseFacet = props.i18n.t('collapse-facet', {label});
  const clearFilters = props.i18n.t('clear-filters', {
    count: props.numberOfActiveValues,
  });
  const clearFiltersForFacet = props.i18n.t('clear-filters-for-facet', {
    count: props.numberOfActiveValues,
    label,
  });

  return html`
    ${renderButton({
      props: {
        part: 'label-button',
        style: 'text-transparent',
        class:
          'flex w-full justify-between rounded-none px-2 py-1 text-lg font-bold',
        ariaLabel: props.isCollapsed ? expandFacet : collapseFacet,
        onClick: props.onToggleCollapse,
        ariaExpanded: props.isCollapsed ? 'false' : 'true',
        ref: (el) => props.headerRef?.(el as HTMLButtonElement),
      },
    })(
      html`${renderHeading({
          props: {level: props.headingLevel, class: 'truncate'},
        })(html`${label}`)}
        <atomic-icon
          part="label-button-icon"
          class="ml-4 w-3 shrink-0 self-center"
          .icon=${props.isCollapsed ? ArrowBottomIcon : ArrowTopIcon}
        ></atomic-icon>`
    )}
    ${when(
      props.onClearFilters && props.numberOfActiveValues > 0,
      () =>
        html`${renderButton({
          props: {
            part: 'clear-button',
            style: 'text-primary',
            class: 'flex max-w-full items-baseline p-2 text-sm',
            ariaLabel: clearFiltersForFacet,
            onClick: props.onClearFilters,
          },
        })(
          html`<atomic-icon
              part="clear-button-icon"
              class="mr-1 h-2 w-2"
              .icon=${CloseIcon}
            ></atomic-icon>
            <span>${clearFilters}</span>`
        )}`,
      () => nothing
    )}
  `;
};
