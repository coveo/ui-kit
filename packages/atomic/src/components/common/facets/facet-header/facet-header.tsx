import {FunctionalComponent, h} from '@stencil/core';
import {i18n} from 'i18next';
import ArrowBottomIcon from '../../../../images/arrow-bottom-rounded.svg';
import ArrowTopIcon from '../../../../images/arrow-top-rounded.svg';
import CloseIcon from '../../../../images/close.svg';
import {Button} from '../../button';
import {Heading} from '../../heading';

export interface FacetHeaderProps {
  i18n: i18n;
  label: string;
  numberOfActiveValues: number;
  isCollapsed: boolean;
  headingLevel: number;
  onToggleCollapse(): void;
  onClearFilters?(): void;
  headerRef?: (element?: HTMLButtonElement) => void;
}

export const FacetHeader: FunctionalComponent<FacetHeaderProps> = (props) => {
  const label = props.i18n.t(props.label);
  const expandFacet = props.i18n.t('expand-facet', {label});
  const collapseFacet = props.i18n.t('collapse-facet', {label});
  const clearFilters = props.i18n.t('clear-filters', {
    count: props.numberOfActiveValues,
  });
  const clearFiltersForFacet = props.i18n.t('clear-filters-for-facet', {
    count: props.numberOfActiveValues,
    label,
  });

  return [
    <Button
      style="text-transparent"
      part="label-button"
      class="flex font-bold justify-between w-full py-1 px-2 text-lg rounded-none"
      title={props.isCollapsed ? expandFacet : collapseFacet}
      onClick={() => props.onToggleCollapse()}
      ariaExpanded={(!props.isCollapsed).toString()}
      ref={props.headerRef}
    >
      <Heading level={props.headingLevel} class="truncate">
        {label}
      </Heading>
      <atomic-icon
        part="label-button-icon"
        class="w-3 self-center shrink-0 ml-4"
        icon={props.isCollapsed ? ArrowBottomIcon : ArrowTopIcon}
      ></atomic-icon>
    </Button>,
    props.onClearFilters && props.numberOfActiveValues > 0 && (
      <Button
        style="text-primary"
        part="clear-button"
        class="flex items-baseline max-w-full p-2 text-sm"
        ariaLabel={clearFiltersForFacet}
        onClick={() => props.onClearFilters!()}
      >
        <atomic-icon
          part="clear-button-icon"
          class="w-2 h-2 mr-1"
          icon={CloseIcon}
        ></atomic-icon>
        <span>{clearFilters}</span>
      </Button>
    ),
  ];
};
