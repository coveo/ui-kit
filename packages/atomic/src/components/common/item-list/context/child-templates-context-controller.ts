import {ItemTemplateProvider} from '@/src/components/common/item-list/item-template-provider';
import {LitElementWithError} from '@/src/decorators/types';
import {buildCustomEvent} from '@/src/utils/event-utils';
import {ReactiveController, ReactiveControllerHost} from 'lit';

const childTemplatesContextEventName = 'atomic/resolveChildTemplates';

/**
 * A reactive controller that manages child template context data from parent components.
 * Handles fetching ItemTemplateProvider via custom events and manages the template provider state.
 */
export class ChildTemplatesContextController implements ReactiveController {
  private host: ReactiveControllerHost & LitElementWithError;
  private _itemTemplateProvider: ItemTemplateProvider | null = null;

  constructor(host: ReactiveControllerHost & LitElementWithError) {
    this.host = host;
    host.addController(this);
  }

  get itemTemplateProvider(): ItemTemplateProvider | null {
    return this._itemTemplateProvider;
  }

  hostConnected(): void {
    this._resolveChildTemplatesContext();
  }

  hostUpdated(): void {
    if (!this._itemTemplateProvider) {
      this._resolveChildTemplatesContext();
    }
  }

  private _resolveChildTemplatesContext(): void {
    const event = buildCustomEvent(
      childTemplatesContextEventName,
      (itemTemplateProvider?: ItemTemplateProvider) => {
        if (this._itemTemplateProvider) {
          return;
        }

        this._itemTemplateProvider = itemTemplateProvider || null;
        this.host.requestUpdate();
      }
    );

    const canceled = this.host.dispatchEvent(event);
    if (canceled) {
      this._itemTemplateProvider = null;
      this.host.requestUpdate();
    }
  }
}

type ChildTemplatesContextEventHandler = (
  itemTemplateProvider?: ItemTemplateProvider
) => void;
export type ChildTemplatesContextEvent =
  CustomEvent<ChildTemplatesContextEventHandler>;
