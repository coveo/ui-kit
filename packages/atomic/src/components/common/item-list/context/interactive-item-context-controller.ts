import {AnyItem} from '@/src/components/common/interface/item.js';
import {LitElementWithError} from '@/src/decorators/types';
import {buildCustomEvent} from '@/src/utils/event-utils';
import {ReactiveController, ReactiveControllerHost} from 'lit';

const interactiveItemContextEventName = 'atomic/resolveInteractiveResult';

/**
 * A reactive controller that manages interactive item context data from parent components.
 * Handles fetching interactive item data via custom events.
 */
export class InteractiveItemContextController<T extends AnyItem = AnyItem>
  implements ReactiveController
{
  private host: ReactiveControllerHost & LitElementWithError;
  private _interactiveItem: T | null = null;

  constructor(host: ReactiveControllerHost & LitElementWithError) {
    this.host = host;
    host.addController(this);
  }

  get interactiveItem(): T | null {
    return this._interactiveItem;
  }

  hostConnected(): void {
    this._resolveInteractiveItemContext();
  }

  private _resolveInteractiveItemContext(): void {
    const event = buildCustomEvent(
      interactiveItemContextEventName,
      (item: AnyItem) => {
        this._interactiveItem = item as T;
        this.host.requestUpdate();
      }
    );

    this.host.dispatchEvent(event);
  }
}

export type InteractiveItemContextEvent<T extends AnyItem = AnyItem> =
  CustomEvent<(interactiveItem: T) => void>;
