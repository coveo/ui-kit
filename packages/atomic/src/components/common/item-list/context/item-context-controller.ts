import type {LitElement, ReactiveController, ReactiveControllerHost} from 'lit';
import type {InitializableComponent} from '@/src/decorators/types.js';
import {buildCustomEvent} from '@/src/utils/event-utils';

type LitElementWithError = Omit<
  Pick<InitializableComponent, 'error'>,
  'error'
> &
  LitElement & {
    error: Error | null;
  };

const itemContextEventName = 'atomic/resolveResult';

export class MissingParentError extends Error {
  constructor(elementName: string, parentName: string) {
    super(
      `The "${elementName}" element must be the child of an "${parentName}" element.`
    );
  }
}

function extractFolded<T = Record<string, unknown>>(
  item: Record<string, unknown>,
  returnFolded: boolean
): T {
  if (returnFolded) {
    if ('children' in item) {
      return item as T;
    } else {
      return {children: [], result: item} as T;
    }
  }

  if ('children' in item && 'result' in item) {
    return item.result as T;
  }
  return item as T;
}

/**
 * A reactive controller that manages item context data from parent components.
 * Handles fetching item data via custom events and manages error states.
 */
export class ItemContextController<T = Record<string, unknown>>
  implements ReactiveController
{
  private host: ReactiveControllerHost & LitElementWithError;
  private parentName: string;
  private folded: boolean;
  private _item: T | null = null;
  private _error: MissingParentError | null = null;

  constructor(
    host: ReactiveControllerHost & LitElementWithError,
    options: {parentName?: string; folded?: boolean} = {}
  ) {
    this.host = host;
    this.parentName = options.parentName ?? 'atomic-result';
    this.folded = options.folded ?? false;
    host.addController(this);
  }

  get item(): T | null {
    return this._error ? null : this._item;
  }

  get error(): MissingParentError | null {
    return this._error;
  }

  get hasError(): boolean {
    return this._error !== null;
  }

  hostConnected(): void {
    this._resolveItemContext();
  }

  private _resolveItemContext(): void {
    const event = buildCustomEvent(
      itemContextEventName,
      (item: Record<string, unknown>) => {
        this._item = extractFolded<T>(item, this.folded);
        this._error = null;
        this.host.error = null;
        this.host.requestUpdate();
      }
    );

    const canceled = this.host.dispatchEvent(event);
    if (canceled) {
      const elementName = (this.host as Element).nodeName.toLowerCase();
      this._error = new MissingParentError(elementName, this.parentName);
      this._item = null;
      this.host.error = this._error;
      this.host.requestUpdate();
    }
  }
}

type ItemContextEventHandler<T> = (item: T) => void;
export type ItemContextEvent<T> = CustomEvent<ItemContextEventHandler<T>>;
