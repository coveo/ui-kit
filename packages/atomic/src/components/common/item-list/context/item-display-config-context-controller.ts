import type {LitElement, ReactiveController, ReactiveControllerHost} from 'lit';
import type {
  ItemDisplayDensity,
  ItemDisplayImageSize,
} from '@/src/components/common/layout/display-options';
import {buildCustomEvent} from '@/src/utils/event-utils';

const itemDisplayConfigContextEventName = 'atomic/resolveResultDisplayConfig';

export type DisplayConfig = {
  density: ItemDisplayDensity;
  imageSize: ItemDisplayImageSize;
};

/**
 * A reactive controller that manages display configuration context from parent components.
 * Handles fetching display config data via custom events.
 */
export class ItemDisplayConfigContextController implements ReactiveController {
  private host: ReactiveControllerHost & LitElement;
  private _config: DisplayConfig | null = null;

  constructor(host: ReactiveControllerHost & LitElement) {
    this.host = host;
    host.addController(this);
  }

  get config(): DisplayConfig | null {
    return this._config;
  }

  hostConnected(): void {
    this._resolveDisplayConfig();
  }

  hostUpdated(): void {
    this._resolveDisplayConfig();
  }

  private _resolveDisplayConfig(): void {
    const event = buildCustomEvent(
      itemDisplayConfigContextEventName,
      (config: DisplayConfig) => {
        this._config = config;
        this.host.requestUpdate();
      }
    );

    const canceled = this.host.dispatchEvent(event);
    if (canceled) {
      this._config = null;
    }
  }
}

type ItemDisplayConfigContextEventHandler = (config: DisplayConfig) => void;
export type ItemDisplayConfigContextEvent =
  CustomEvent<ItemDisplayConfigContextEventHandler>;
