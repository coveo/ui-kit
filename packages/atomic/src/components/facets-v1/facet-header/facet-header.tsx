import {FunctionalComponent, h} from '@stencil/core';
import ArrowTopIcon from 'coveo-styleguide/resources/icons/svg/arrow-top-rounded.svg';
import ArrowBottomIcon from 'coveo-styleguide/resources/icons/svg/arrow-bottom-rounded.svg';
import CloseIcon from 'coveo-styleguide/resources/icons/svg/close.svg';
import {i18n} from 'i18next';

export const FacetHeader: FunctionalComponent<{
  i18n: i18n;
  label: string;
  numberOfSelectedValues: number;
  isCollapsed: boolean;
  onToggleCollapse(): void;
  onClearFilters(): void;
}> = (props) => {
  const label = props.i18n.t(props.label);
  const expandFacet = props.i18n.t('expandFacet', {label});
  const collapseFacet = props.i18n.t('collapseFacet', {label});
  const clearFilters = props.i18n.t('clearFilters', {
    count: props.numberOfSelectedValues,
  });
  const clearFiltersForFacet = props.i18n.t('clearFiltersForFacet', {
    count: props.numberOfSelectedValues,
    label,
  });

  return [
    <button
      part="label-button"
      class="flex justify-between w-full py-1 text-on-background text-lg hover:text-primary"
      title={props.isCollapsed ? expandFacet : collapseFacet}
      onClick={() => props.onToggleCollapse()}
    >
      <span class="ellipsed">{label}</span>
      <span
        part="label-button-icon"
        class="fill-current w-3 h-2 self-center flex-shrink-0 ml-4"
        innerHTML={props.isCollapsed ? ArrowTopIcon : ArrowBottomIcon}
      ></span>
    </button>,
    props.numberOfSelectedValues > 0 && (
      <button
        part="clear-button"
        class="flex w-full p-1 text-secondary hover:text-secondary-light text-sm"
        title={clearFiltersForFacet}
        onClick={() => props.onClearFilters()}
      >
        <span
          part="clear-button-icon"
          class="fill-current w-2 h-2 self-center mr-1"
          innerHTML={CloseIcon}
        ></span>
        <span>{clearFilters}</span>
      </button>
    ),
  ];
};
