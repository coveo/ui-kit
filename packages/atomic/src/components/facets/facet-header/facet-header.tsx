import {FunctionalComponent, h} from '@stencil/core';
import ArrowTopIcon from 'coveo-styleguide/resources/icons/svg/arrow-top-rounded.svg';
import ArrowBottomIcon from 'coveo-styleguide/resources/icons/svg/arrow-bottom-rounded.svg';
import CloseIcon from 'coveo-styleguide/resources/icons/svg/close.svg';
import {i18n} from 'i18next';
import {Button} from '../../common/button';

export const FacetHeader: FunctionalComponent<{
  i18n: i18n;
  label: string;
  numberOfSelectedValues: number;
  isCollapsed: boolean;
  onToggleCollapse(): void;
  onClearFilters?(): void;
}> = (props) => {
  const label = props.i18n.t(props.label);
  const expandFacet = props.i18n.t('expand-facet', {label});
  const collapseFacet = props.i18n.t('collapse-facet', {label});
  const clearFilters = props.i18n.t('clear-filters', {
    count: props.numberOfSelectedValues,
  });
  const clearFiltersForFacet = props.i18n.t('clear-filters-for-facet', {
    count: props.numberOfSelectedValues,
    label,
  });

  return [
    <Button
      style="text-transparent"
      part="label-button"
      class="flex font-bold justify-between w-full py-1 px-2 text-lg"
      title={props.isCollapsed ? expandFacet : collapseFacet}
      onClick={() => props.onToggleCollapse()}
      ariaExpanded={(!props.isCollapsed).toString()}
      text={label}
    >
      <atomic-icon
        part="label-button-icon"
        class="w-3 self-center flex-shrink-0 ml-4"
        icon={props.isCollapsed ? ArrowTopIcon : ArrowBottomIcon}
      ></atomic-icon>
    </Button>,
    props.onClearFilters && props.numberOfSelectedValues > 0 && (
      <Button
        style="text-primary"
        part="clear-button"
        class="flex items-baseline max-w-full p-2 text-sm"
        title={clearFiltersForFacet}
        onClick={() => props.onClearFilters!()}
      >
        <atomic-icon
          part="clear-button-icon"
          class="w-2 h-2 mr-1"
          icon={CloseIcon}
        ></atomic-icon>
        <span>{clearFilters}</span>
      </Button>
    ),
  ];
};
