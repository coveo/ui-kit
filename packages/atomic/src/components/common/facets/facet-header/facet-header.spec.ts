import {renderFunctionFixture} from '@/vitest-utils/testing-helpers/fixture';
import {createTestI18n} from '@/vitest-utils/testing-helpers/i18n-utils';
import {page} from '@vitest/browser/context';
import '@vitest/browser/matchers.d.ts';
import {html} from 'lit';
import {vi, expect, describe, it, beforeAll} from 'vitest';
import {renderFacetHeader, FacetHeaderProps} from './facet-header';

describe('facetHeader', () => {
  let i18n: Awaited<ReturnType<typeof createTestI18n>>;

  const locators = {
    get clearButton() {
      return page.getByRole('button', {name: 'Clear 3 filters'});
    },
    get clearButtonIcon() {
      return locators.clearButton.element().querySelector('atomic-icon');
    },
    get collapseButton() {
      return page.getByRole('button', {name: 'Collapse the Test Facet facet'});
    },
    get collapseButtonIcon() {
      return locators.collapseButton.element().querySelector('atomic-icon');
    },
    get expandButton() {
      return page.getByRole('button', {name: 'Expand the Test Facet facet'});
    },
    get expandButtonIcon() {
      return locators.expandButton.element().querySelector('atomic-icon');
    },
  };

  beforeAll(async () => {
    i18n = await createTestI18n();
  });

  const renderComponent = async (props: Partial<FacetHeaderProps> = {}) => {
    const defaultProps: FacetHeaderProps = {
      i18n: i18n,
      label: 'Test Facet',
      numberOfActiveValues: 3,
      isCollapsed: false,
      headingLevel: 2,
      onToggleCollapse: vi.fn(),
      onClearFilters: vi.fn(),
      headerRef: vi.fn(),
    };
    const mergedProps = {...defaultProps, ...props};
    return renderFunctionFixture(
      html`${renderFacetHeader({props: mergedProps})}`
    );
  };

  it('renders the clear button with the correct aria label attribute', async () => {
    await renderComponent();
    await expect.element(locators.clearButton).toBeInTheDocument();
    await expect
      .element(locators.clearButton)
      .toHaveAttribute(
        'aria-label',
        'Clear 3 filters for the Test Facet facet'
      );
  });

  it('renders the label button with the correct aria label attribute', async () => {
    await renderComponent();
    await expect.element(locators.collapseButton).toBeInTheDocument();
    await expect
      .element(locators.collapseButton)
      .toHaveAttribute('aria-label', 'Collapse the Test Facet facet');
  });

  it('renders the label button with aria expanded set to true', async () => {
    await renderComponent();
    await expect.element(locators.collapseButton).toBeInTheDocument();
    await expect
      .element(locators.collapseButton)
      .toHaveAttribute('aria-expanded', 'true');
  });

  it('renders the label button with aria expanded set to false', async () => {
    await renderComponent({isCollapsed: true});
    await expect.element(locators.expandButton).toBeInTheDocument();
    await expect
      .element(locators.expandButton)
      .toHaveAttribute('aria-expanded', 'false');
  });

  it('renders the expand icon when the facet is collapsed', async () => {
    await renderComponent({isCollapsed: true});
    await expect.element(locators.expandButtonIcon).toBeInTheDocument();
    expect(locators.expandButtonIcon?.getAttribute('icon')).toMatchSnapshot();
  });

  it('renders the collapse icon when the facet is expanded', async () => {
    await renderComponent({isCollapsed: false});
    await expect.element(locators.collapseButtonIcon).toBeInTheDocument();
    expect(locators.collapseButtonIcon?.getAttribute('icon')).toMatchSnapshot();
  });

  it('renders the collapse filters button when there are active values', async () => {
    await renderComponent();
    await expect.element(locators.collapseButton).toBeInTheDocument();
  });

  it('calls #onToggleCollapse when the label button is clicked', async () => {
    const onToggleCollapseMock = vi.fn();
    await renderComponent({
      onToggleCollapse: onToggleCollapseMock,
    });
    locators.collapseButton.element().dispatchEvent(new Event('click'));
    expect(onToggleCollapseMock).toHaveBeenCalled();
  });

  it('should have the correct part attribute on the label button', async () => {
    await renderComponent();
    await expect
      .element(locators.collapseButton)
      .toHaveAttribute('part', 'label-button');
  });

  it('should have the correct part attribute on the label button icon', async () => {
    await renderComponent();
    await expect
      .element(locators.collapseButtonIcon)
      .toHaveAttribute('part', 'label-button-icon');
  });

  it('should have the correct part attribute on the clear button', async () => {
    await renderComponent();
    await expect
      .element(locators.clearButton)
      .toHaveAttribute('part', 'clear-button');
  });

  it('should have the correct part attribute on the clear button icon', async () => {
    await renderComponent();
    await expect
      .element(locators.clearButtonIcon)
      .toHaveAttribute('part', 'clear-button-icon');
  });

  it('does not render the clear filters button when there are no active values', async () => {
    await renderComponent({numberOfActiveValues: 0});
    await expect.element(locators.clearButton).not.toBeInTheDocument();
  });

  it('calls onClearFilters when the clear filters button is clicked', async () => {
    const onClearFiltersMock = vi.fn();
    await renderComponent({
      onClearFilters: onClearFiltersMock,
    });
    await locators.clearButton.element().dispatchEvent(new Event('click'));
    expect(onClearFiltersMock).toHaveBeenCalled();
  });

  it('renders the heading with the correct level and label', async () => {
    await renderComponent();
    const heading = page.getByText('Test Facet');
    await expect.element(heading).toBeInTheDocument();
    expect(heading.element().tagName.toLowerCase()).toBe('h2');
  });

  it('calls headerRef with the button element', async () => {
    const headerRefMock = vi.fn();
    await renderComponent({headerRef: headerRefMock});
    expect(headerRefMock).toHaveBeenCalledWith(expect.any(HTMLButtonElement));
  });
});
