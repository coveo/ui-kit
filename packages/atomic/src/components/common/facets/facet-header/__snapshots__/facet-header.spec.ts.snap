// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`#renderFacetHeader > renders the collapse icon when the facet is expanded 1`] = `"<svg viewBox="0 0 12.6 7.2" xmlns="http://www.w3.org/2000/svg"><path d="m11.3 7.04c-.3 0-.5-.1-.7-.3l-4.6-4.6-4.6 4.6c-.4.4-1 .4-1.4 0s-.4-1 0-1.4l5.2-5.2c.4-.4 1.2-.4 1.6 0l5.2 5.2c.4.4.4 1 0 1.4-.2.2-.4.3-.7.3"/></svg>"`;

exports[`#renderFacetHeader > renders the expand icon when the facet is collapsed 1`] = `"<svg viewBox="0 0 12.6 7.2" xmlns="http://www.w3.org/2000/svg"><path d="m11.421 7.04c-.3 0-.5-.1-.7-.3l-4.6-4.6-4.6 4.6c-.4.4-1 .4-1.4 0s-.4-1 0-1.4l5.2-5.2c.4-.4 1.2-.4 1.6 0l5.2 5.2c.4.4.4 1 0 1.4-.2.2-.4.3-.7.3" transform="matrix(-1 0 0 -1 12.366 7.086)"/></svg>"`;

exports[`facetHeader > renders the collapse icon when the facet is expanded 1`] = `"<svg viewBox="0 0 12.6 7.2" xmlns="http://www.w3.org/2000/svg"><path d="m11.3 7.04c-.3 0-.5-.1-.7-.3l-4.6-4.6-4.6 4.6c-.4.4-1 .4-1.4 0s-.4-1 0-1.4l5.2-5.2c.4-.4 1.2-.4 1.6 0l5.2 5.2c.4.4.4 1 0 1.4-.2.2-.4.3-.7.3"/></svg>"`;

exports[`facetHeader > renders the expand icon when the facet is collapsed 1`] = `"<svg viewBox="0 0 12.6 7.2" xmlns="http://www.w3.org/2000/svg"><path d="m11.421 7.04c-.3 0-.5-.1-.7-.3l-4.6-4.6-4.6 4.6c-.4.4-1 .4-1.4 0s-.4-1 0-1.4l5.2-5.2c.4-.4 1.2-.4 1.6 0l5.2 5.2c.4.4.4 1 0 1.4-.2.2-.4.3-.7.3" transform="matrix(-1 0 0 -1 12.366 7.086)"/></svg>"`;
